restapi/admin_users.go

```diff
@@ -467,3 +474,50 @@ func getAddUsersListToGroupsResponse(session *models.Principal, params admin_api
 
 	return nil
 }
+
+func getListUsersWithAccessToBucketResponse(session *models.Principal, bucket string) ([]string, *models.Error) {
+	ctx := context.Background()
+	mAdmin, err := newMAdminClient(session)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	// create a minioClient interface implementation
+	// defining the client to be used
+	adminClient := adminClient{client: mAdmin}
+
+	users, err := listUsers(ctx, adminClient)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	var retval []string
+	seen := make(map[string]bool)
+	for i := 0; i < len(users); i++ {
+		policy, err := adminClient.getPolicy(ctx, users[i].Policy)
+		if err == nil {
+			parsedPolicy, err2 := parsePolicy(users[i].Policy, policy)
+			if err2 == nil && policyMatchesBucket(parsedPolicy, bucket) {
+				retval = append(retval, users[i].AccessKey)
+				seen[users[i].AccessKey] = true
+			}
+		}
```

Comment:
We should log errors in case we need to debug in the feature

```suggestion
		if err != nil {
		    // log error
	            continue    
		}
		parsedPolicy, err := parsePolicy(users[i].Policy, policy)
		if err != nil {
		    // log error
	            continue    
		}
		if policyMatchesBucket(parsedPolicy, bucket) {
			retval = append(retval, users[i].AccessKey)
			seen[users[i].AccessKey] = true
		}
```